pkg/unvanquished_src.dpkdir/ui/lua/circle_menu.lua

```diff
@@ -86,12 +93,32 @@ function CircleMenu:HandleKey(event)
 		return
 	end
 	if index <= #self.items then
-		-- event:StopImmediatePropagation()
 		event:StopPropagation()
 		self.menuElement.child_nodes[index+1].first_child:Click()
 	end
 end
 
+function CircleMenu:Quit(event)
```

Comment:
What's with this super complicated quit thing? It worked fine before without that.

Reply:
>It worked fine before without that.
Not exactly, there was a lot of corner cases of undesired behavior around clicking and click-releasing on buttons and backdrop, closing menu when it shouldn't and leaving it open when it should.

Managing events from click events is a bit tricky, especially since `onclick`, `onmouseup` and `onmousedown` are separate. I also found that one click event on body element tends to leak out and fire **after** menu document is closed, no matter what I do.
And like I mentioned that in comments, trying to hide document already hidden (and deallocated, I presume) does, on occasion, crush the program.

Reply:
I wonder if this is due to us using this `Events.push_event` hack instead of calling the methods directly. I think Events.push_event() causes the action to occur on the next frame instead of immediately.

So, in most places where you do `Events.pushevent("hide foo")`, you could probably do `self.document:Blur()`

This should cause the hide action to take effect immediately.